github-action.py

```diff
@@ -740,6 +750,226 @@ def upload(self):
+                config=self.config,
+                manager=self.manager,
+                iso_timestamp=self.iso_timestamp,
+            )
+
+    def notify_build_status_on_timeout(self):
+        self.notify_build_status("failed", additional_info="Timeout")
+
+    def notify_build_status(
+        self, status, stage="build", log_file=None, additional_info=None
+    ):
+        notify_issues_cmd = [
+            f"{str(PROJECT_PATH)}/utils/notify-issues",
+            f"--message-templates-dir={str(PROJECT_PATH)}/templates",
+        ]
+
+        if log_file:
+            notify_issues_cmd += [
+                f"--build-log={self.get_build_log_url(log_file=log_file)}"
+            ]
+
+        if additional_info:
+            notify_issues_cmd += [f"--additional-info={str(additional_info)}"]
+
+        package_name = f"iso-{self.dist.name}-{self.iso_version}"
+
+        notify_issues_cmd += [
+            stage,
+            self.qubes_release,
+            str(self.builder_dir),
+            package_name,
+            self.dist.distribution,
+            status,
+        ]
+
+        try:
+            subprocess.run(notify_issues_cmd, env=self.env)
+        except subprocess.CalledProcessError as e:
+            msg = f"{package_name}: Failed to notify GitHub: {str(e)}"
+            log.error(msg)
+
+    def notify_upload_status(self, log_file=None, additional_info=None):
+        notify_issues_cmd = [
+            f"{str(PROJECT_PATH)}/utils/notify-issues",
+            f"--message-templates-dir={str(PROJECT_PATH)}/templates",
+        ]
+
+        if log_file:
+            notify_issues_cmd += [
+                f"--build-log={self.get_build_log_url(log_file=log_file)}"
+            ]
+
+        if additional_info:
+            notify_issues_cmd += [f"--additional-info={str(additional_info)}"]
+
+        package_name = f"iso-{self.dist.name}-{self.iso_version}"
+
+        state_file = (
+            self.state_dir
+            / f"{self.qubes_release}-iso-{self.dist.name}"
+            # type: ignore
+        )
+        stable_state_file = (
+            self.state_dir
+            / f"{self.qubes_release}-iso-{self.dist.name}-current"
+            # type: ignore
+        )
+        notify_issues_cmd += [
+            "upload",
+            self.qubes_release,
+            str(self.builder_dir),
+            package_name,
+            self.dist.distribution,
+            str(self.repository_publish),
+            str(state_file),
+            str(stable_state_file),
+        ]
+        if self.iso_base_url:
+            notify_issues_cmd += [
+                "--repository-url",
+                self.iso_base_url / self.repository_publish,
+            ]
+
+        try:
+            subprocess.run(notify_issues_cmd, env=self.env)
+        except subprocess.CalledProcessError as e:
+            msg = f"{package_name}: Failed to notify GitHub: {str(e)}"
+            log.error(msg)
+
+    def upload(self):
+        raise NotImplemented
```

Comment:
```suggestion
        raise NotImplementedError
```
`NotImplemented` exists too, but is not an exception